### PR TITLE
Update Terragrunt to support Terraform 0.9

### LINKED ISSUE #158: How Terragrunt Will Handle Terraform 0.9.0 Compatibility
Hello Terragrunt Community,

As of today, Terragrunt is not compatible with Terraform 0.9.x. We want to fix that. This GitHub Issue will outline our current thinking on how we plan to do that.

- **Handling Remote State in Terraform 0.9:** Unfortunately, Terraform 0.9.x removed the `terraform remote config` commands, which is why Terragrunt is not usable with Terraform 0.9.x. To resolve this, we’ve been exploring the best way to configure remote state in Terraform 0.9.x.

   Ideally, Terragrunt could simply retire this feature altogether, as Terraform 0.9 allows you to define your remote state configuration in your .tf files. Unfortunately, Terraform does not support all the features Terragrunt had, such as  `path_relative_to_include()`, manual lock controls, and automatic retry when a lock cannot be obtained.

   Our current thinking is to move to Terraform’s remote state configuration, but with variables where you’d like to use Terragrunt features:

   ```hcl
   # In some .tf file
   terraform {
      backend "s3" {
         bucket = "my-bucket"
         key    = "${var.remote_state_key}"
         region = "us-east-1"
      }
   }
   ```

   You can then configure Terragrunt to pass in the variables, using the interpolation helpers as before:

   ```hcl
   # In terraform.tfvars or .terragrunt
   terragrunt = {
      terraform {
         extra_arguments "remote_state" {
            arguments = [
                "-var", "remote_state_key=${path_relative_to_include()}",
            ]
            commands = [
               "apply",
               "plan",
               "import",
               "push",
               "refresh"
            ]
          }
       }
   }
   ```

- **User Migration Strategy:** Many of our users (and Gruntwork clients) have multiple folders configured to use Terragrunt. We plan to explicitly think through a recommended migration strategy. This could be a simple recommendation to manually make updates, or some kind of automation. We’re still thinking through this.

- **Handling Retries When a Lock Cannot be Obtained:** When Terragrunt cannot obtain a lock, it keeps trying to get one until it times out. Currently, Terraform 0.9.x fails immediately if a lock cannot be obtained. Some CI scripts may depend on having an automatic retry, so we intend to wrap the native Terraform backends code with our own lock-retry mechanism.

### Timeline

In an ideal world, we’d begin these changes right away, but we are currently at capacity with other work. We will start making fixes to Terragrunt as soon as we can, but that may not be for another week or two. If anyone wishes to submit a PR, we would be grateful! 

### Feedback

Feedback is always welcome. Please feel free to leave additional comments in this thread. Thanks for using Terragrunt, and for your patience while we add support for Terraform 0.9.x!

### PR DESCRIPTION
**Pull request by brikis98**
This PR updates Terragrunt so it works with Terraform 0.9.x. This finally fixes #158.

Here’s the approach I used:

1. Instead of calling `terraform remote config` (which was removed in Terraform 0.9), we now call `terraform init` and pass `-backend-config` parameters as necessary.

1. I have removed support for DynamoDB locking, since this is now built into Terraform itself. To enable it, you just specify a `lock_table` attribute in your s3 backend configuration. I’ve updated Terragrunt to automatically create this table if it doesn’t exist already. If you want to Terraform to retry locks, you can use an `extra_arguments` block.

1. I’ve completely refactored the docs to emphasize that Terragrunt is now focused not on locking and remote state, but working with multiple modules and keeping your Terraform code DRY.

I’ve included a section in the README on how to migrate from the old Terragrunt and Terraform to this new version. Feedback is welcome!

It may be easier to browse the README directly in the branch: https://github.com/gruntwork-io/terragrunt/tree/terraform-0.9. You can find the migration instructions here: https://github.com/gruntwork-io/terragrunt/tree/terraform-0.9#migrating-from-terragrunt-v011x-and-terraform-08x-and-older.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,636 +1,539 @@
...
+Terragrunt has the ability to download remote Terraform configurations. The idea is that you define the Terraform code 
+for all of your infrastructure in a single repo, called, for example, `modules`:
 
-[terragrunt] 2016/05/27 00:39:19 Attempting to release lock for state file my-app in DynamoDB
-[terragrunt] 2016/05/27 00:39:19 Lock released!
+```
+└── modules
+    ├── prod
+    │   ├── app
+    │   │   └── main.tf
```
Comment (tomstockton) [inline README.md]: Doesn't this pattern still end up with a lot of copy and paste between the resources defined in `prod/app/main.tf` and `otherenv/app/main.tf`?  Does the `modules` repo actually need to be environment specific?

Comment (brikis98) [inline README.md]: Whoops! Good catch. That's a copy/paste error. I've updated that example with the proper folder structure for the `modules` repo: https://github.com/gruntwork-io/terragrunt/pull/167/commits/ef92f6e7c47e4f2c90eb2e0f18293b7f0bc7e730.

**Comment by tomstockton** [comment]
How does this handle environment specific remote state files?  I imagine this would be best done using [state environments](https://www.terraform.io/docs/state/environments.html) but I didn't notice it in the code.

**Comment by brikis98** [comment]
> How does this handle environment specific remote state files? 

What are you referring to with "this"?

**Comment by tomstockton** [comment]
I'm referring to this change / PR.

**Comment by brikis98** [comment]
I'm not sure this PR has any impact on remote state files... Can you give an example of what you mean?

**Comment by tomstockton** [comment]
I've reread the README, let me rewrite my initial question better!

Are you considering using the Terraform native support for [state environments](https://www.terraform.io/docs/state/environments.html) or is it not needed due to how the remote statefile path is derived using `"${path_relative_to_include()}/terraform.tfstate"`?

**Comment by brikis98** [comment]
I think Terraform's new state environments are a completely alternative way to manage multiple environments from what we describe in the Terragrunt README. The Terragrunt approach is to define all those environments in files on disk so that you can read the files, version them, revert them, etc. The state environments approach is to run a command to pick a different environment and perhaps to put conditional statements throughout your code so it behaves differently based on which environment was selected. I find the former approach easier to understand and less error prone, but AFAIK, nothing in Terragrunt or this PR prevents you from using the latter approach.

**Comment by tomstockton** [comment]
Thanks for the clarification.  I was slightly confused about the use case for the state environments as it seemed to conflict with your 'Golden Rules' of Terraform!  Good to hear your thoughts on it.

File: README.md [hunk 2]
```diff
@@ -643,340 +531,322 @@ terragrunt = {
         "refresh"
       ]
     }
+  }
+}
+```
```
Comment (josh-padnick) [inline README.md]: I had to read this a few times to get it. I think the word "switch" through me. Maybe reword? Also, I didn't immediately make the connection you were still using a list of strings but had the first and second string as arg and arg-value.

File: README.md [hunk 3]
```diff
@@ -643,340 +531,322 @@ terragrunt = {
...
 ```
 
-### The apply-all, destroy-all, and output-all commands
 
-Let's say you have a single environment (e.g. `stage` or `prod`) that has a number of Terraform modules within it:
+
+
+
+
+### Work with multiple Terraform modules
```
Comment (josh-padnick) [inline README.md]: Maybe better wording here would be "Quickly deploy multiple Terraform modules."

File: README.md [hunk 4]
```diff
@@ -643,340 +531,322 @@ terragrunt = {
...
+Inside each `terraform.tfvars` file, add a `terragrunt = { ... }` block to identify this as a module managed by 
+Terragrunt (the block can be empty or include any of the configs described in this documentation):
+
+```hcl
+terragrunt = {
+  # Put your Terragrunt configuration here
+}
+```
+
+Now you can go into the root folder and deploy all the modules within it by using the `apply-all` command:
```
Comment (josh-padnick) [inline README.md]: Can you clarify what you mean by "root folder". If there are multiple environments, each with their own folder tree, does "root" mean any folder that contains one or more subfolders that contain` terraform.tfvars` files?

File: README.md [hunk 5]
```diff
@@ -643,340 +531,322 @@ terragrunt = {
...
 ```
-cd my-terraform-repo/stage
+cd root
 terragrunt apply-all
 ```
 
-When you run this command, Terragrunt will find all `terraform.tfvars` files in the subfolders of the current working 
-directory that contain `terragrunt = { ... }` blocks, and run `terragrunt apply` in each one concurrently.
+When you run this command, Terragrunt will look through all the subfolders of the current working directory, find all 
+`terraform.tfvars` files with a `terragrunt = { ... }` block, and run `terragrunt apply` in each one concurrently.
```
Comment (josh-padnick) [inline README.md]: Ok, I guess you what root means here...

Comment (josh-padnick) [inline README.md]: But what about recursive subfolders?

File: README.md [hunk 6]
```diff
@@ -643,340 +531,322 @@ terragrunt = {
...
   dependencies {
     paths = ["../vpc", "../backend-app"]
   }
 }
 ```
 
-Once you've specified the depenedencies in each `terraform.tfvars` file, when you run the `terragrunt apply-all` and 
+Once you've specified the dependencies in each `terraform.tfvars` file, when you run the `terragrunt apply-all` or 
 `terragrunt destroy-all`, Terragrunt will ensure that the dependencies are applied or destroyed, respectively, in the
 correct order. For the example at the start of this section, the order for the `apply-all` command would be:
```
Comment (josh-padnick) [inline README.md]: ...while applying or destroying Terraform modules in parallel where possible.

File: README.md [hunk 7]
```diff
@@ -643,340 +531,322 @@ terragrunt = {
...
+as the environment variable `TF_VAR_foo` and to read that value in using this `get_env` helper function.
 
-## CLI Options
+
+
+
+### CLI Options
 
 Terragrunt forwards all arguments and options to Terraform. The only exceptions are the options that start with the
 prefix `--terragrunt-`. The currently available options are:
```
Comment (josh-padnick) [inline README.md]: The other exception is `--version`.

File: README.md [hunk 8]
```diff
@@ -643,340 +531,322 @@ terragrunt = {
...
 
 * `--terragrunt-config`: A custom path to the `terraform.tfvars` file. May also be specified via the `TERRAGRUNT_CONFIG`
   environment variable. The default path is `terraform.tfvars` in the current directory (see 
   [Terragrunt config files](#terragrnt-config-files) for a slightly more nuanced explanation).
```
Comment (josh-padnick) [inline README.md]: Does not apply with `apply-all`, etc. subcommands.

File: README.md [hunk 9]
```diff
@@ -643,340 +531,322 @@ terragrunt = {
...
+
 * `--terragrunt-working-dir`: Set the directory where Terragrunt should execute the `terraform` command. Default is the
   current working directory. Note that for the `apply-all` and `destroy-all` directories, this parameter has a different 
   meaning: Terragrunt will apply or destroy all the Terraform modules in the subfolders of the 
   `terragrunt-working-dir`, running `terraform` in the root of each module it finds.
+
 * `--terragrunt-source`: Download Terraform configurations from the specified source into a temporary folder, and run 
   Terraform in that temporary folder. May also be specified via the `TERRAGRUNT_SOURCE` environment variable. The 
   source should use the same syntax as the [Terraform module source](https://www.terraform.io/docs/modules/sources.html) 
   parameter.
```
Comment (josh-padnick) [inline README.md]: Does not apply with apply-all, etc. subcommands.

File: README.md [hunk 10]
```diff
@@ -1039,7 +911,122 @@ To migrate, all you need to do is:
...
+          "untaint"
+        ]
+      
+        arguments = [
+          "-lock-timeout=20m"
+        ]    
+      }     
+    }     
+    ``` 
+
```
Comment (josh-padnick) [inline README.md]: These migration instructions are great: two simple steps.

File: circle.yml [hunk 11]
```diff
@@ -7,8 +7,7 @@ dependencies:
     # Install the gruntwork-module-circleci-helpers and use it to configure the build environment and run tests.
     - curl -Ls https://raw.githubusercontent.com/gruntwork-io/gruntwork-installer/master/bootstrap-gruntwork-installer.sh | bash /dev/stdin --version 0.0.13
```
Comment (josh-padnick) [inline circle.yml]: Upgrade gruntwork-installer to latest version.

File: cli/cli_app.go [hunk 12]
```diff
@@ -111,18 +99,13 @@ func CreateTerragruntCli(version string, writer io.Writer, errwriter io.Writer)
...
 	app.Writer = writer
 	app.ErrWriter = errwriter
-	app.UsageText = fmt.Sprintf(`Terragrunt is a thin wrapper for [Terraform](https://www.terraform.io/) that supports locking
-   via Amazon's DynamoDB and enforces best practices. Terragrunt forwards almost all commands, arguments, and options
-   directly to Terraform, using whatever version of Terraform you already have installed. However, before running
-   Terraform, Terragrunt will ensure your remote state is configured according to the settings in %s.
-   Moreover, for the apply and destroy commands, Terragrunt will first try to acquire a lock using DynamoDB. For
-   documentation, see https://github.com/gruntwork-io/terragrunt/.`, config.DefaultTerragruntConfigPath)
+	app.UsageText = `Terragrunt is a thin wrapper for Terraform that provides extra tools for working with multiple
+   Terraform modules, remote state, and locking. For documentation, see https://github.com/gruntwork-io/terragrunt/.`
```
Comment (josh-padnick) [inline cli/cli_app.go]: Nice update.

File: dynamodb/dynamo_lock_table.go [hunk 13]
```diff
@@ -15,6 +16,26 @@ import (
 // running many automated tests in parallel, we use a counting semaphore
 var tableCreateDeleteSemaphore = NewCountingSemaphore(10)
 
+// Terraform requires the DynamoDB table to have a primary key with this name
+const ATTR_LOCK_ID = "LockID"
```
Comment (josh-padnick) [inline dynamodb/dynamo_lock_table.go]: What are the implications of this with multiple Terraform/Terragrunt modules each writing a lock? Does that mean that otherwise parallelizable Terraform modules can no longer be run in parallel?

File: dynamodb/dynamo_lock_table_test.go [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+		waitGroup.Add(1)
+		go func() {
+			defer waitGroup.Done()
+			err := CreateLockTableIfNecessary(tableName, client, mockOptions)
+			assert.Nil(t, err, "Unexpected error: %v", err)
+		}()
+	}
+
+	waitGroup.Wait()
+}
```
Comment (josh-padnick) [inline dynamodb/dynamo_lock_table_test.go]: Nice test.

File: remote/remote_state_s3.go [hunk 15]
```diff
@@ -14,11 +15,12 @@ import (
...
-	Bucket  string
-	Key     string
-	Region  string
-	Profile string
+	Encrypt   string `mapstructure:"encrypt"`
+	Bucket    string `mapstructure:"bucket"`
+	Key       string `mapstructure:"key"`
+	Region    string `mapstructure:"region"`
+	Profile   string `mapstructure:"profile"`
+	LockTable string `mapstructure:"lock_table"`
```
Comment (josh-padnick) [inline remote/remote_state_s3.go]: Just observing that it's a little weird that `LockTable` is part of the `RemoteStateConfigS3` struct (versus something like a `RemoteStateConfigAWS` struct, but still makes sense.

**Comment by brikis98** [comment]
> One thought I had is that after reading everything, the reader still has to make one more leap on their own: What is the ideal, complete setup with Terragrunt that combines all these use cases? For that reason, it'd be nice if we had a _sample folder with maybe live and modules subfolders which mimicked a real setup and showed exactly what the root terraform.tfvars file and each "child" terraform.tfvars file looks like.

That's a *great* idea, but I think it's outside the scope of this PR. Could you file a separate bug to track this?

File: README.md [hunk 16]
```diff
@@ -1,636 +1,524 @@
...
+  state storage and the `bucket` you specify in `remote_state.config` doesn't already exist, Terragrunt will create it 
+  automatically, with [versioning enabled](http://docs.aws.amazon.com/AmazonS3/latest/dev/Versioning.html). 
+
+* **DynamoDB table**: If you are using the [S3 backend](https://www.terraform.io/docs/backends/types/s3.html) for 
+  remote state storage and you specify a `lock_table` (a [DynamoDB table used for 
+  locking](https://www.terraform.io/docs/backends/types/s3.html#lock_table)) in `remote_state.config`, if that table
+  doesn't already exist, Terragrunt will create it automatically, including a primary key called `LockID`.
+
+**Note*: If you specify a `profile` key in `remote_state.config`, Terragrunt will automatically use this AWS profile 
+when creating the S3 bucket or DynamoDB table.
```
Comment (brikis98) [inline README.md]: My preference is to keep all Terraform code (`.tf`) in the `modules` repo, even if it's only used in one environment. Versioning and consistency are both good things and the amount of overhead this adds, especially with the `--terragrunt-source` param, is fairly minor if you're already using this for everything else.

File: README.md (hunk 6 above)
Comment (brikis98) [inline README.md]: Sorry, what does this refer to?

**Comment by josh-padnick** [comment]
> That's a *great* idea, but I think it's outside the scope of this PR. Could you file a separate bug to track this?

Thanks! https://github.com/gruntwork-io/terragrunt/issues/169

File: README.md [hunk 17]
```diff
@@ -1,636 +1,524 @@
...
+}
 
-#### Remote state management configuration
+instance_count = 3
+instance_type = "t2.micro"
+```
 
-For remote state management, Terragrunt supports the following settings in `terraform.tfvars`:
+*(Note: the double slash (`//`) is intentional and required. It's part of Terraform's Git syntax for [module 
+sources](https://www.terraform.io/docs/modules/sources.html).)
```
Comment (brikis98) [inline README.md]: Good catch. Fixed: https://github.com/gruntwork-io/terragrunt/pull/167/commits/5af50b0d64c2cf113a35458edbac78085597401c

File: README.md [hunk 18]
```diff
@@ -1,636 +1,524 @@
...
+environment-specific values in the `.tfvars` files, which is about as close to the "essential complexity" of the 
+problem as you can get.
 
-* Includes
-* Find parent helper
-* Relative path helper
-* Overriding included settings
-* The `apply-all`, `destroy-all`, and `output-all` commands
-* Dependencies between modules
+Just as importantly, since the Terraform code is now defined in a single repo, you can version it (e.g., using Git
```
Comment (brikis98) [inline README.md]: Fixed: https://github.com/gruntwork-io/terragrunt/pull/167/commits/5af50b0d64c2cf113a35458edbac78085597401c

File: README.md [hunk 19]
```diff
@@ -1,636 +1,524 @@
...
+│   └── terraform.tfvars
+└── vpc
+    ├── main.tf
+    └── terraform.tfvars
+```
 
-You can read in environment variables in the `terragrunt = { ... }` portion of your `terraform.tfvars` file using the 
-`get_env()` helper function:
+In your root `terraform.tfvars` file, you can define your entire remote state configuration just once in a 
+`remote_state` block, as follows:
```
Comment (brikis98) [inline README.md]: Fixed: https://github.com/gruntwork-io/terragrunt/pull/167/commits/5af50b0d64c2cf113a35458edbac78085597401c

File: README.md [hunk 20]
```diff
@@ -1,636 +1,524 @@
...
-is mandatory but can be empty (e.g. `${get_env("ENVIRONMENT_VARIABLE_NAME", "")}`).
+The `remote_state` block supports all the same [backend types](https://www.terraform.io/docs/backends/types/index.html) 
+as Terraform. The next time you run `terragrunt`, it will automatically configure all the settings in the
+`remote_state.config` block, if they aren't configured already, by calling [terraform 
+init](https://www.terraform.io/docs/commands/init.html).
 
-If there is no environment variable with that name registered in the system, the configuration file would be evaluated 
-to:
+In each of the child `terraform.tfvars` files, such as `mysql/terraform.tfvars`, you can tell Terragrunt to 
+automatically include all the settings from the root `terraform.tfvars` file as follows:
```
Comment (brikis98) [inline README.md]: Fixed: https://github.com/gruntwork-io/terragrunt/pull/167/commits/5af50b0d64c2cf113a35458edbac78085597401c

File: README.md [hunk 21]
```diff
@@ -1,636 +1,524 @@
...
-      bucket = "value-bucket"
-      key = "/foo/bar/terraform.tfstate"
-      region = "us-west-2"
-    }
-  }
-}
-```
+* `path_relative_to_include()`: This helper function returns the relative path between the current `terraform.tfvars` 
+  file and the path specified in its `include` block. We use this so that each Terraform module stores its Terraform
+  state at a different `key`. For example, the `mysql` module will have its `key` parameter resolve to
```
Comment (brikis98) [inline README.md]: Fixed: https://github.com/gruntwork-io/terragrunt/pull/167/commits/5af50b0d64c2cf113a35458edbac78085597401c

File: README.md (hunk 16 above)
Comment (brikis98) [inline README.md]: Good catch. Fixed: https://github.com/gruntwork-io/terragrunt/pull/167/commits/5af50b0d64c2cf113a35458edbac78085597401c

File: README.md [hunk 22]
```diff
@@ -1,636 +1,524 @@
...
+    extra_arguments "retry_lock" {
+      commands = [
+        "init",
+        "apply",
+        "refresh",
+        "import",
+        "plan",
+        "taint",
+        "untaint"
+      ]
```
Comment (brikis98) [inline README.md]: > This list of terraform commands that apply to locks is non-obvious. It'd be nice if we could another helper command here that generates this array for you.

This is another good candidate for a separate issue, if you don't mind filing it.

> BTW, where does retry_lock come from? I didn't see it in Terraform docs after a quick search, so I'm guessing it's an arbitrary var name? Would be good to clarify either way.

Arbitrary name. I updated the docs to clarify: https://github.com/gruntwork-io/terragrunt/pull/167/commits/5af50b0d64c2cf113a35458edbac78085597401c

File: README.md (hunk 2 above)
Comment (brikis98) [inline README.md]: Updated the docs to make this clearer: https://github.com/gruntwork-io/terragrunt/pull/167/commits/5af50b0d64c2cf113a35458edbac78085597401c

File: README.md (hunk 3 above)
Comment (brikis98) [inline README.md]: Well, it also allows you to destroy and run output for multiple modules, so I'll stick with "work with".

File: README.md (hunk 4 above)
Comment (brikis98) [inline README.md]: The root here was literally the folder named root in example folder structure above. I added syntax highlighting to the word to clarify that: https://github.com/gruntwork-io/terragrunt/pull/167/commits/5af50b0d64c2cf113a35458edbac78085597401c

File: README.md (hunk 5 above)
Comment (brikis98) [inline README.md]: That includes recursive subfolders. I added the word "recursive" to clarify: https://github.com/gruntwork-io/terragrunt/pull/167/commits/5af50b0d64c2cf113a35458edbac78085597401c

File: README.md (hunk 7 above)
Comment (brikis98) [inline README.md]: Added: https://github.com/gruntwork-io/terragrunt/pull/167/commits/5af50b0d64c2cf113a35458edbac78085597401c

File: README.md (hunk 8 above)
Comment (brikis98) [inline README.md]: Added: https://github.com/gruntwork-io/terragrunt/pull/167/commits/5af50b0d64c2cf113a35458edbac78085597401c

File: README.md (hunk 9 above)
Comment (brikis98) [inline README.md]: Added: https://github.com/gruntwork-io/terragrunt/pull/167/commits/5af50b0d64c2cf113a35458edbac78085597401c

File: README.md (hunk 16 above)
Comment (josh-padnick) [inline README.md]: Consider the use case of adding Route53 Public Hosted Zones. So your position would be to write a trivial Terraform module that does the following:

```hcl
resource "aws_route53_zone" "primary" {
  name = "${var.domain_name}"
}
```

...so that you could have the following Terragrunt files in the `live` repo:

```hcl
terragrunt = {
  include {
    path = "${find_in_parent_folders()}"
  }
  terraform {
    source = "git::git@github.com:foo/modules.git//route53_hosted_zone?ref=v0.0.3"
  }
}

domain_name = "example.com"
# Alternatively
domain_names = ["example.com"]
```

I guess you're arguing the standardization gives us the `apply-all` command, and forces any complexity into a versionable module. In this extreme trivial case, the overhead outweighs the value but still gives us consistency.

I'd say the one potential area is this may break down a little is if Terraform's HCL isn't expressive enough to capture what we want. For example, I often wish modules accepted lists of lists, but since we can only pass "flat" lists, we may have to instantiate one module per each "thing".

Mostly thinking out loud here...

File: circle.yml (hunk 11 above)
Comment (brikis98) [inline circle.yml]: Updated: https://github.com/gruntwork-io/terragrunt/pull/167/commits/854f00e57f939d6980fe62aa7ce117a8a2d4190f

File: dynamodb/dynamo_lock_table.go (hunk 13 above)
Comment (brikis98) [inline dynamodb/dynamo_lock_table.go]: `LockID` is the name of the column in DynamoDB. The value that gets written to it is the `key` parameter of the S3 bucket (perhaps with the S3 bucket name prefixed, I forget). Therefore, it's a unique lock for each unique state file.

File: remote/remote_state_s3.go (hunk 15 above)
Comment (brikis98) [inline remote/remote_state_s3.go]: It's part of the S3 backend definition: https://www.terraform.io/docs/backends/types/s3.html#lock_table

**Comment by brikis98** [comment]
Merging!

File: README.md (hunk 22 above)
Comment (josh-padnick) [inline README.md]: https://github.com/gruntwork-io/terragrunt/issues/171

Comment (brikis98) [inline README.md]: Thx!
